src/main/java/ui/components/issuepicker/IssuePickerDialog.java

```diff
@@ -0,0 +1,250 @@
+        selectedIssues.getChildren().clear();
+        Map<String, FlowPane> repoContent = getRepoContent(chosenIssues, currentSuggestion);
+
+        repoContent.entrySet().forEach(entry -> {
+            entry.getValue().getChildren().add(0, createRepoTitle(entry.getKey()));
+            selectedIssues.getChildren().addAll(entry.getValue());
+        });
+    }
+
+    private void populateSuggestedIssues(List<TurboIssue> matchedIssues, Optional<TurboIssue> currentSuggestion) {
+        suggestedIssues.getChildren().clear();
+        matchedIssues.stream()
+            .forEach(issue -> suggestedIssues.getChildren().add(processSuggestedIssue(issue, currentSuggestion)));
+    }
+
+    private Node processSelectedIssue(TurboIssue issue, Optional<TurboIssue> suggestion) {
+        PickerIssue styledIssue = new PickerIssue(issue);
+
+        if (suggestion.isPresent() && suggestion.get().equals(issue)) {
+            return styledIssue.faded(true).removed(true).getNode();
+        }
+        return new PickerIssue(issue).getNode();
+    }
+
+    private Node processSuggestedIssue(TurboIssue issue, Optional<TurboIssue> currentSuggestion) {
+        GuiElement element = new GuiElement(issue, models.getLabelsOfIssue(issue), models.getMilestoneOfIssue(issue),
+                                            models.getAssigneeOfIssue(issue), models.getAuthorOfIssue(issue));
+        IssueCard card = new IssueCard(element, isSuggestedIssue(issue, currentSuggestion));
+        card.setOnMouseClicked(e -> handleIssueClick(issue, card));
+        return card;
+    }
+
+
+    // Event handling 
+
+    /**
+     * Updates state of the issue picker based on the entire query
+     */
+    private final void handleUserInput(String query) {
+        state = new IssuePickerState(allIssues, query.toLowerCase());
+        populatePanes(state);
+    }
+    
+    private void handleIssueClick(TurboIssue issue, IssueCard card) {
+        issuepickerQueryField.setDisable(true);
+        Platform.runLater(card::requestFocus);
+        state.updateSelectedIssues(issue);
+        populateSelectedIssues(state.getSelectedIssues(), state.getCurrentSuggestion());
+    }
+
+    // UI helper methods
+
+    private void createMainLayout() {
+        FXMLLoader loader = new FXMLLoader(UI.class.getResource(FXML_PATH));
+        loader.setController(this);
+        try {
+            issuepickerLayout = (VBox) loader.load();
+        } catch (IOException e) {
+            logger.error("Failure to load FXML. " + e.getMessage());
+            close();
+        }
+    }
+
+    private void createButtons() {
+        ButtonType confirmButtonType = new ButtonType("Confirm", ButtonData.OK_DONE);
+        getDialogPane().getButtonTypes().addAll(confirmButtonType, ButtonType.CANCEL);
+
+        // defines what happens when user confirms/presses enter
+        setResultConverter(dialogButton -> {
+            if (dialogButton == confirmButtonType) {
```

Comment:
Is it possible that we can have multiple OK_DONE buttons?

But anyway, what I was saying was that you might be able to use `equals` instead of `==`, which seems a bit fragile (depends on implementation details).